cmd/kpod/push.go

```diff
@@ -0,0 +1,220 @@
+	dockerRegistryOptions
+	// SigningOptions encapsulates settings that control whether or not we
+	// strip or add signatures to the image when pushing (uploading) the
+	// image to a registry.
+	signingOptions
+}
+
+func pushCmd(c *cli.Context) error {
+	var registryCreds *types.DockerAuthConfig
+
+	args := c.Args()
+	if len(args) < 2 {
+		return errors.New("kpod push requires exactly 2 arguments")
+	}
+	srcName := c.Args().Get(0)
+	destName := c.Args().Get(1)
+
+	signaturePolicy := ""
+	if c.IsSet("signature-policy") {
+		signaturePolicy = c.String("signature-policy")
+	}
+	compress := archive.Uncompressed
+	if !c.IsSet("disable-compression") || !c.Bool("disable-compression") {
+		compress = archive.Gzip
+	}
+	quiet := false
+	if c.IsSet("quiet") {
+		quiet = c.Bool("quiet")
+	}
+	registryCredsString := ""
+	if c.IsSet("creds") {
+		registryCredsString = c.String("creds")
+	}
+	certPath := ""
+	if c.IsSet("cert-dir") {
+		certPath = c.String("cert-dir")
+	}
+	skipVerify := false
+	if c.IsSet("tls-verify") {
+		skipVerify = !c.BoolT("tls-verify")
+	}
+	removeSignatures := false
+	if c.IsSet("remove-signatures") {
+		removeSignatures = c.Bool("remove-signatures")
+	}
+	signBy := ""
+	if c.IsSet("sign-by") {
+		signBy = c.String("sign-by")
+	}
+
+	if registryCredsString != "" {
+		creds, err := parseRegistryCreds(registryCredsString)
+		if err != nil {
+			return err
+		}
+		registryCreds = creds
+	}
+
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	options := pushOptions{
+		Compression:         compress,
+		SignaturePolicyPath: signaturePolicy,
+		Store:               store,
+		dockerRegistryOptions: dockerRegistryOptions{
+			DockerRegistryCreds:         registryCreds,
+			DockerCertPath:              certPath,
+			DockerInsecureSkipTLSVerify: skipVerify,
+		},
+		signingOptions: signingOptions{
+			RemoveSignatures: removeSignatures,
+			SignBy:           signBy,
+		},
+	}
+	if !quiet {
+		options.ReportWriter = os.Stderr
+	}
+	return pushImage(srcName, destName, options)
+}
+
+func pushImage(srcName, destName string, options pushOptions) error {
+	if srcName == "" || destName == "" {
+		return errors.Errorf("source and destination image names must be specified")
+	}
+
+	// Get the destination Image Reference
+	dest, err := alltransports.ParseImageName(destName)
+	if err != nil {
+		return errors.Wrapf(err, "error getting destination imageReference for %q", destName)
+	}
+
+	policyContext, err := getPolicyContext(options.SignaturePolicyPath)
+	if err != nil {
+		return err
+	}
+	defer policyContext.Destroy()
+	// Look up the image name and its layer, then build the imagePushData from
+	// the image
+	img, err := findImage(options.Store, srcName)
+	if err != nil {
+		return errors.Wrapf(err, "error locating image %q for importing settings", srcName)
+	}
+	systemContext := getSystemContext(options.SignaturePolicyPath)
+	ipd, err := importImagePushDataFromImage(options.Store, img, systemContext)
+	if err != nil {
+		return err
```

Comment:
It does, the errors are wrapped in a human-readable message within the function.  Should I wrap them again wherever the function is called to pinpoint the cause of the error further?